service/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateLandingZoneResourcesFlightStep.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.landingzone.stairway.flight.create;
+
+import bio.terra.common.iam.BearerToken;
+import bio.terra.landingzone.job.JobMapKeys;
+import bio.terra.landingzone.service.landingzone.azure.LandingZoneService;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneRequest;
+import bio.terra.landingzone.stairway.flight.LandingZoneFlightMapKeys;
+import bio.terra.profile.model.ProfileModel;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.FlightNotFoundException;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class CreateLandingZoneResourcesFlightStep implements Step {
+
+  private final LandingZoneService landingZoneService;
+  private final LandingZoneRequest landingZoneRequest;
+  private final UUID landingZoneId;
+  private final String jobIdKey;
+
+  public CreateLandingZoneResourcesFlightStep(
+      LandingZoneService landingZoneService,
+      LandingZoneRequest landingZoneRequest,
+      // TODO: check why this is not in landingZoneRequest
+      UUID landingZoneId,
+      String jobIdKey) {
+    this.landingZoneService = landingZoneService;
+    this.landingZoneRequest = landingZoneRequest;
+    this.landingZoneId = landingZoneId;
+    this.jobIdKey = jobIdKey;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var subflightId = getSubflightId(landingZoneId.toString());
+
+    if (isFlightAlreadyExists(context, subflightId)) {
```

Comment:
This is interesting. What situation specifically is this guarding against?

Reply:
I think step should be idempotent and retry should be successful.

Reply:
Are you saying that this means the step was previously called and is still in the process of retrying? What if it was called previously but failed-- wouldn't the flight exist but be in a non-successful state?

Reply:
I mean that Stairway steps may run multiple times (not because of retry, but if I am not mistaken it doesn't support run-once semantic), so we may already have started this flight. If it was failed, then I think the flight will be failed too.